src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class Figure {
+    private static final int MAX_NUMBER = 10;
+    private Random random = new Random();
+
+    public String getColor() {
+        ColorSupplier color = new ColorSupplier();
+        return color.getRandomColor();
+    }
+
+    public int getNumber() {
+        return random.nextInt(MAX_NUMBER);
+    }
+
+    public abstract double getArea();
+
+    public abstract String toString();
```

Comment:
the correct way is to create two separate interfaces with these methods 
then we should implement these interfaces in the Figure class and override in Circle, Rectuncle,... classes